### PR TITLE
relion - added new stable version 2.1

### PR DESCRIPTION
**Pull request by JusticeForMikeBrown**
This is now the recommended version to use.

### CODE REVIEW

File: var/spack/repos/builtin/packages/relion/package.py [hunk 1]
```diff
@@ -34,11 +34,9 @@ class Relion(CMakePackage):
...
     url      = "https://github.com/3dem/relion"
 
-    version('2.1-beta-1', git='https://github.com/3dem/relion.git',
-            commit='e7607a869687b636d3c39e0d5b6a9cba930fc3b2')
-    version('2.0.3', git='https://github.com/3dem/relion.git',
-            preferred='true')
-
+    version('2.0.3', git='https://github.com/3dem/relion.git', tag='2.0.3')
+    version('2.1', git='https://github.com/3dem/relion.git', tag='2.1',
+        preferred='true')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/relion/package.py]: no need for `preferred`, it will be chosen automatically.

Comment (JusticeForMikeBrown) [reply var/spack/repos/builtin/packages/relion/package.py]: Because it's the higher number?

Comment (davydden) [reply var/spack/repos/builtin/packages/relion/package.py]: yes, also please use descending order. That's how we it's done in most packages.

Comment (JusticeForMikeBrown) [reply var/spack/repos/builtin/packages/relion/package.py]: Done!

**Pull request merged**